From 5114385b4040a5f8fe12251825131d5c033e752d Mon Sep 17 00:00:00 2001
From: Jan Tulak <jtulak@redhat.com>
Date: Thu, 21 Feb 2019 15:38:13 +0100
Subject: [PATCH 01/22] ssm: use LC_ALL=C when running tools from ssm

Previously, we used LC_NUMERIC, but if LC_ALL is set by the user/system,
then LC_NUMERIC is ignored, leading to issues with locales with
different decimal delimiter literal than a dot. So, set LC_ALL instead
of LC_NUMERIC.

This forces other localized things like error messages to English. But
because ssm does not have translations, we can have errors reported by
tools (like lvm) within ssm always in English as well.

Resolves: rhbz#1679587
Resolves: rhbz#1679591
Resolves: #7 (github)

Signed-off-by: Jan Tulak <jtulak@redhat.com>
Reviewed-by: Lukas Czerner <lczerner@redhat.com>
---
 ssmlib/main.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ssmlib/main.py b/ssmlib/main.py
index 6d9f9f0..a148e85 100644
--- a/ssmlib/main.py
+++ b/ssmlib/main.py
@@ -31,7 +31,7 @@ EXTN = ['ext2', 'ext3', 'ext4']
 SUPPORTED_FS = ['xfs', 'btrfs'] + EXTN
 SUPPORTED_BACKENDS = ['lvm', 'btrfs', 'crypt']
 SUPPORTED_RAID = ['0', '1', '10']
-os.environ['LC_NUMERIC'] = "C"
+os.environ['LC_ALL'] = "C"
 
 # If you change this please change doc/conf.py as well
 VERSION = '0.4'
-- 
2.21.0

